app/src/debug/java/com/artemzin/qualitymatters/network/OkHttpInterceptorsModule.java

```diff
@@ -0,0 +1,43 @@
+package com.artemzin.qualitymatters.network;
+
+import android.support.annotation.NonNull;
+
+import com.facebook.stetho.okhttp3.StethoInterceptor;
+
+import java.util.List;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import dagger.Module;
+import dagger.Provides;
+import okhttp3.Interceptor;
+import okhttp3.logging.HttpLoggingInterceptor;
+import timber.log.Timber;
+
+import static com.artemzin.qualitymatters.network.NetworkModule.LIST_OF_OK_HTTP_INTERCEPTORS;
+import static com.artemzin.qualitymatters.network.NetworkModule.LIST_OF_OK_HTTP_NETWORK_INTERCEPTORS;
+import static java.util.Collections.singletonList;
+
+/**
+ * Provides OkHttp interceptors for debug build.
+ */
+@Module
+public class OkHttpInterceptorsModule {
+
+    // Provided as separate dependency for Developer Settings to be able to change HTTP log level at runtime.
+    @Provides @Singleton @NonNull
+    public HttpLoggingInterceptor provideHttpLoggingInterceptor() {
+        return new HttpLoggingInterceptor(message -> Timber.d(message));
+    }
+
+    @Provides @Named(LIST_OF_OK_HTTP_INTERCEPTORS) @Singleton @NonNull
+    public List<Interceptor> provideOkHttpInterceptors(@NonNull HttpLoggingInterceptor httpLoggingInterceptor) {
+        return singletonList(httpLoggingInterceptor);
+    }
+
+    @Provides @Named(LIST_OF_OK_HTTP_NETWORK_INTERCEPTORS) @Singleton @NonNull
```

Comment:
Ok, will use qualifiers!

On Sun, 7 Feb 2016, 23:19 Niklas Baudy <notifications@github.com> wrote:

> In
> app/src/debug/java/com/artemzin/qualitymatters/network/OkHttpInterceptorsModule.java
> <https://github.com/artem-zinnatullin/qualitymatters/pull/88#discussion_r52124050>
> :
>
> > + */
> > +@Module
> > +public class OkHttpInterceptorsModule {
> > +
> > +    // Provided as separate dependency for Developer Settings to be able to change HTTP log level at runtime.
> > +    @Provides @Singleton @NonNull
> > +    public HttpLoggingInterceptor provideHttpLoggingInterceptor() {
> > +        return new HttpLoggingInterceptor(message -> Timber.d(message));
> > +    }
> > +
> > +    @Provides @Named(LIST_OF_OK_HTTP_
>
> INTERCEPTORS) @Singleton @NonNull
> > +    public List<Interceptor> provideOkHttpInterceptors(@NonNull HttpLoggingInterceptor httpLoggingInterceptor) {
> > +        return singletonList(httpLoggingInterceptor);
> > +    }
> > +
> > +    @Provides @Named(LIST_OF_OK_HTTP_NETWORK_INTERCEPTORS) @Singleton @NonNull
>
> Well you don't need the constants anymore.
> You can't misuse the constants since they don't exist.
> It just reads a bit easier @Provides @OkHttpNetworkInterceptors
> @Singleton @NonNull.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/artem-zinnatullin/qualitymatters/pull/88/files#r52124050>
> .
>
-- 

@artem_zin